dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/liberty/CDIRuntimeImpl.java

```diff
@@ -699,4 +703,15 @@ public BuildCompatibleExtensionFinder getBuildCompatibleExtensionFinder() {
         return bceFinder;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public WeldDevelopmentMode getWeldDevelopmentMode() {
+        WeldDevelopmentMode devMode = null;
+        //only return the service if it exists and is enabled
+        if (this.weldDevelopmentMode != null && this.weldDevelopmentMode.enabled()) {
+            devMode = this.weldDevelopmentMode;
+        }
```

Comment:
Given that `this.weldDevelopmentMode` is volatile and so could change value while this code is running, I think we should copy it to a local variable before checking for `null` and enablement.

Reply:
ok